### DISCUSSION TITLE
发现停止工作是因为refresh_token过期，建议能不能用账号+密码自动获取refresh_token。

### DISCUSSION THREAD

**Discussion by withtea** [messense/aliyundrive-webdav#110] [Ideas]
```
2021-11-02T19:26:00.939177Z ERROR aliyundrive_webdav::drive: refresh token failed: HTTP status client error (400 Bad Request) for url (https://websv.aliyundrive.com/token/refresh): 
<h1>The input parameter refresh_token is not valid. Please refer to document. RequestId:E662145B-4A35-422B-81CA-3783519292EA</h1>
<h2>400</h2>
<h2>ADABFAC1-E697-48AE-8F4D-1A5F3F1D3AC7</h2>
```

Error: initialize aliyundrive client failed
发现停止工作是因为refresh_token过期，建议能不能用账号+密码自动获取refresh_tokeren。不然停止工作只有每隔一段时间手动输入一次新的refresh_tokeren。

**Comment by messense** [comment]
refresh_token 有效期 2 个小时，会自动刷新 refresh_token，除非断网超过 2 个小时，一般不会有问题。

不会加用户名密码登录方式。

**Reply by withtea** [reply]
我出现过2次，网络一直都是连接状态的，是在软路由上，如果断网有系统日志记录。考虑发送刷新refresh_token请求timeout没有retry的可能性吗？

**Reply by messense** [reply]
已经 retry 了 10 次。

https://github.com/messense/aliyundrive-webdav/blob/e6c6d61fc7da323e875537b83f6b0ab8ffd5a920/src/drive/mod.rs#L167

**Reply by withtea** [reply]
好的，如有更多log，我再来反馈。

**Reply by thjw81** [reply]
在openwrt出现好几次连不上阿里云盘，才明白是refresh_token会发生变化。填上最新的就好了。一直连接一点问题没有。可能是中断一段时间，这个token就失效了。

**Comment by upsetterlesson** [comment]
梅林上不会自动刷新refresh_token我只填了refresh_token

**Comment by upsetterlesson** [comment]
很快就过期了

**Comment by wangshengxin1998** [comment]
能不能把最新的refresh_token储存起来,进程意外重启的时候能够直接使用最新的refresh_token

**Comment by jimmylzt188** [comment]
阿里的token经常变，能否增加扫码登录云盘获取token？
